src/lib/converter/plugins/GitHubPlugin.ts

```diff
@@ -82,9 +90,13 @@ class Repository {
             });
         }
 
-        out = <ShellJS.ExecOutputReturnValue> ShellJS.exec('git rev-parse --short HEAD', {silent: true});
-        if (out.code === 0) {
-            this.branch = out.stdout.replace('\n', '');
+        if (!this.revision) {
+            out = <ShellJS.ExecOutputReturnValue> ShellJS.exec('git rev-parse --short HEAD', {silent: true});
+            if (out.code === 0) {
+                this.branch = out.stdout.replace('\n', '');
+            }
+        } else if (this.revision) {
+            this.branch = this.revision;
```

Comment:
Hm. I mean it's already declared as `branch` in [src/lib/converter/plugins/GitHubPlugin.ts#L33](https://github.com/TypeStrong/typedoc/blob/master/src/lib/converter/plugins/GitHubPlugin.ts#L33). Basically it can be a `branch` or a `revision` and until now it was declared as branch first and when it finds a git repository it replaces it with the revision. I don't mind `reference` either but `branch` or `revision` feels more natural to me.

(`revision` actually makes the most sense to me because a branch can be a revsion and well a revision is a revision 😆  - `branch` can just be a branch and not a revision from a logical point of view)